support/mock/mockrt.go

```diff
@@ -496,6 +496,33 @@ func (rt *Runtime) Reset() {
 	rt.expectCreateActor = nil
 }
 
+func (rt *Runtime) ExpectAssertionFailure(expected string, f func()) {
+	prevState := rt.state
+
+	defer func() {
+		r := recover()
+		if r == nil {
+			rt.t.Errorf("expected panic with message %v but call succeeded", expected)
+			return
+		}
+		a, ok := r.(abort)
+		if ok {
+			rt.t.Errorf("expected panic with message %v but got abort %v", expected, a)
+			return
+		}
+		p, ok := r.(string)
+		if !ok {
+			panic(r)
+		}
+		if p != expected {
```

Comment:
I think being explicit on the expected reason for an assertion failure is appropriate here since calls to `AssertMsg` are infrequent.